pkgs/top-level/all-packages.nix

```diff
@@ -168,12 +207,51 @@ with pkgs;
 
   fetchDockerLayer = callPackage ../build-support/fetchdocker/fetchDockerLayer.nix { };
 
+  /* `fetchfossil` fetches a project using the fossil version control system
+     (https://www.fossil-scm.org)
+
+     Example:
+        fetchfossil {
+          name = "example"
+          url = "http://example.com";
+          rev = "2eea7584d57b80c053c5e9c5041a8b8185f566f23debe2cb80b83dfc1579c1d9"
+          sha256 = "18dyk2apmnjapd604a5njfqwjri1mc7lgjaajy9phicpibgdrwzh";
+        };
+  */
   fetchfossil = callPackage ../build-support/fetchfossil { };
 
+  /* `fetchgit` fetches a project using the git version control system
+     (https://git-scm.com/)
+
+     Options:
+      - url: URL of the repo
+      - rev: Revision to fetch, uses HEAD when not provided
+      - md5: optional md5 hash
+      - sha256: optional sha256 hash
+      - leaveDotGit: Whether to leave the .git folder in the resulting derivation
+      - fetchSubmodules: Whether to also fetch submodules
+      - deepClone: Clone the entire repository
+      - branchName: Name of the branch to fetch
+
+     Example:
+        fetchgit {
+          url = "git://git.ardour.org/ardour/ardour.git";
+          rev = "ae0dcdc0c5d13483271065c360e378202d20170a";
+          sha256 = "0mla5lm51ryikc2rrk53max2m7a5ds6i1ai921l2h95wrha45nkr";
+        };
+  */
```

Comment:
It seems like the best place for this documentation would be in `../build-support/fetchgit/default.nix` so that it can be maintained along with the actual functions, no?  That decreases the risk of the documentation drifting out of sync with the implementation.

https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/fetchgit/default.nix

Reply:
:thumbsup: